File: docs/examples/local-agent.py
"""Local mode — AI agent answers calls, no cloud needed."""
import asyncio
from patter import Patter


async def main():
    phone = Patter(
        mode="local",
        twilio_sid="AC...",
        twilio_token="...",
        openai_key="sk-...",
        phone_number="+1...",
        webhook_url="xxx.ngrok-free.dev",
    )

    agent = phone.agent(
        system_prompt="You are a friendly customer service agent for Acme Corp.",
        voice="alloy",
        first_message="Hello! Thanks for calling Acme. How can I help?",
    )

    print("Listening for calls...")
    await phone.serve(agent=agent, port=8000)


asyncio.run(main())

File: docs/examples/local-agent.ts
/**
 * Local mode — AI agent answers calls, no cloud needed.
 */
import { Patter } from "patter";

async function main() {
  const phone = new Patter({
    mode: "local",
    twilioSid: "AC...",
    twilioToken: "...",
    openaiKey: "sk-...",
    phoneNumber: "+1...",
    webhookUrl: "xxx.ngrok-free.dev",
  });

  const agent = phone.agent({
    systemPrompt: "You are a friendly customer service agent for Acme Corp.",
    voice: "alloy",
    firstMessage: "Hello! Thanks for calling Acme. How can I help?",
  });

  console.log("Listening for calls...");
  await phone.serve({ agent, port: 8000 });
}

main().catch(console.error);
